### PR TITLE
🎉 Set MAX_RETRIES and MAX_SYNC_JOB_ATTEMPTS by env vars

### PR DESCRIPTION
**Pull request by vladimir-remar**
## What
Having 3 retries by default does not work for us in some cases: in some connectors, the fact of doing a retry after a failure, for example having exceeded a rate limit, makes this retry useless, since a rate limit makes it unfeasible to follow the synchronization, for that reason we would like to define the number of Attemps and retries that is done in a synchronization. Additionally, we are orchestrating our Airbyte jobs through an Airflow DAG, and we can set up custom retry policies per each Airbyte job.

## How
Defining the values of `MAX_SYNC_JOB_ATTEMPTS` and` MAX_RETRIES_PER_ATTEMPT` by environment variables, thus being able to establish control over the attempts.

## Notes
I am aware that this change is not trivial and since I am not familiar with the Java environment I may have left something aside, however the tests by setting  `MAX_SYNC_JOB_ATTEMPTS=2` and  `MAX_RETRIES_PER_ATTEMP=2` carried out with a dev environment give positive results.
Attached log and captures
Captura de pantalla de 2021-07-30 11-21-28
[log.txt](https://github.com/airbytehq/airbyte/files/6907093/log.txt)

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: .env.dev [hunk 1]
```diff
@@ -20,3 +20,5 @@ HACK_LOCAL_ROOT_PARENT=/tmp
 WEBAPP_URL=http://localhost:8000/
 API_URL=/api/v1/
 INTERNAL_API_HOST=airbyte-server:8001
+MAX_RETRIES_PER_ATTEMPT=3
+MAX_SYNC_JOB_ATTEMPTS=3
```
Comment (davinchia) [inline .env.dev:24]: nit: newline

**Review by davinchia — APPROVED** [review]
Thank you!

Can you make one small change? I'll merge it in after

**Review by marcosmarxm — APPROVED** [review]
Awesome!

File: .env.dev (hunk 1 above)
Comment (vladimir-remar) [reply .env.dev]: done

**Comment by davinchia** [comment]
One last thing @vladimir-remar - can you run `./gradlew format` for me to make sure the formatting lines up? Our builds do not run with contributor PRs so I want to make sure.

**Comment by vladimir-remar** [comment]
@davinchia Sorry for the delay, I did the format and attach the log.
[log.txt](https://github.com/airbytehq/airbyte/files/6915039/log.txt)

**Pull request merged**